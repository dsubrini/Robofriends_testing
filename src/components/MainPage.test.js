import React from 'react';
import MainPage from './MainPage';
import { shallow } from 'enzyme';

let wrapper;
beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: false
    };
    wrapper = shallow(<MainPage { ...mockProps }/>)
});

it('renders MaiPage without crashing', function () {
    expect(wrapper).toMatchSnapshot();
});

it('filters robots correctly', function () {
    const mockProps2 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 3,
            name: 'John',
            email: 'john@gmail.com'
        }],
        searchField: 'john',
        isPending: false
    };
    const wrapper2 = shallow(<MainPage { ...mockProps2 }/>);
    expect(wrapper2.instance().filteredRobots()).toEqual([{
        id: 3,
        name: 'John',
        email: 'john@gmail.com'
    }]);
});

it('filters robots correctly 2', function () {
    const mockProps3 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 3,
            name: 'John',
            email: 'john@gmail.com'
        }],
        searchField: 'a',
        isPending: false
    };
    const filteredRobots = []
    const wrapper3 = shallow(<MainPage { ...mockProps3 }/>);
    expect(wrapper3.instance().filteredRobots()).toEqual(filteredRobots);
});

it('render isPending correctly', function () {
    const mockProps4 = {
        onRequestRobots: jest.fn(),
        isPending: true
    };
    const wrapper4 = shallow(<MainPage { ...mockProps4 }/>);
});
