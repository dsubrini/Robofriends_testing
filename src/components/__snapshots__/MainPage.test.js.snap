// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MaiPage without crashing 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundry>
      <CardList
        robots={Array []}
      />
    </ErrorBoundry>
  </Scroll>
</div>
`;
